### DISCUSSION TITLE
[mini.jump2d] Can we replicate what folke/flash.nvim does where labels are characters from the jump target words

### DISCUSSION THREAD

**Discussion by GitMurf** [nvim-mini/mini.nvim#1257] [Q&A] [answered]
Let me know if you are unfamiliar or need me to explain further, but I am curious if we can replicate what flash.nvim does where the jump labels actually consist of the spelling of the target word you want to jump to. So more or less, like a search.

```
foo everywhere
bar there
baz here
neovim is awesome
vim is old school
this is cool
```

When I active jump, if I wanted to jump to `bar` then I would type `b` (it would then limit down the choice of words starting with `b` so just `bar` and `baz` would be left. Then I type `a` (which they both have) and finally `r` is the deciding factor and would jump to the word.

Additionally the way `flash` does it is there is an actual "random" final jump character added for each remaining "valid" word but typing the actual matching words characters narrows down the potential matches and at any time you can jump "early" by using the random unique character. So in my simple example above, I could have just started with `b` and for `bar` maybe it assigns the unique jump character as `x` and for `baz` maybe it is `y`... so if I do `x` it jumps to `bar` and if `y` it jumps to `baz` ... OR I can keep typing another letter to filter down the potential matches.

The TLDR of the value it brings is it eliminate some of the thinking to start because you can activate jump and just start typing the word you are trying to get to and then naturally you can jump to it after a couple characters... instead of having random characters always be your jump labels. Additionally it acts as a pseudo search a bit too and often it is valuable to see all the similar words matches as you type it and filter down the potential jump targets.

Thoughts? Does my explanation make sense? Thanks!

**Comment by GitMurf** [comment]
FYI, I found similar discussion from a while ago here: https://github.com/echasnovski/mini.nvim/discussions/253#discussioncomment-6326872

I still am interested to know if there have been updates since then and/or if this feature is possible by creating our own spotter (without having to do something crazy)? Thanks!

**Comment by echasnovski** [comment, marked as answer]
The closest to the described workflow you can get (while using something pre-built) is with [`single_character`](https://github.com/echasnovski/mini.nvim/blob/61e5d46fc0cf71306c51275383767d996f559a60/doc/mini-jump2d.txt#L430) builtin jump options. Here is an [example of how it can be used](https://github.com/echasnovski/mini.nvim/blob/61e5d46fc0cf71306c51275383767d996f559a60/doc/mini-jump2d.txt#L86) (which can be made into a separate mapping; doesn't quite work as a default options for technical reasons).

To read more about optimizing 'mini.jump2d', you can take a look at #1033.

That said, computing labels independently of which text is underneath them is the core design of 'mini.jump2d' which won't be changed. This is mostly because I find an "initiate jumping with a single key press while intentionally limiting choices" approach more straightforward to understand.

**Reply by GitMurf** [reply]
@echasnovski thanks for the reply! Makes sense and one of the reasons I am leveraging mini so heavily right now is to keep it simple and reduce complexity. So your response is inline with what I am looking for anyways if I actually think through it. Thanks!
